memory/persisted_session.go

```diff
@@ -19,14 +19,20 @@ import "github.com/dolthub/go-mysql-server/sql"
 type GlobalsMap = map[string]interface{}
 type InMemoryPersistedSession struct {
 	sql.Session
-	persistedGlobals GlobalsMap
+	persistedGlobals  GlobalsMap
+	incrementValidate func()
```

Comment:
(minor) `validateCallback` is a slightly more self-descriptive name. We are using it to increment a counter in the new test, but the callback function itself isn't coupled to that implementation, so a more generic name makes sense here.